cmd/abapLandscapePortalUpdateAddOnProduct.go

```diff
@@ -0,0 +1,453 @@
+	ZoneId       string `json:"zone_id"`
+}
+
+type reqEntity struct {
+	RequestId string `json:"RequestId"`
+	ZoneId    string `json:"zone_id"`
+	Status    string `json:"Status"`
+	SystemId  string `json:"SystemId"`
+}
+
+type updateAddOnReq struct {
+	ProductName    string `json:"productName"`
+	ProductVersion string `json:"productVersion"`
+}
+
+type updateAddOnResp struct {
+	RequestId string `json:"requestId"`
+	ZoneId    string `json:"zoneId"`
+	Status    string `json:"status"`
+	SystemId  string `json:"systemId"`
+}
+
+func abapLandscapePortalUpdateAddOnProduct(config abapLandscapePortalUpdateAddOnProductOptions, telemetryData *telemetry.CustomData) {
+	client := http.Client{}
+
+	// Error situations should be bubbled up until they reach the line below which will then stop execution
+	// through the log.Entry().Fatal() call leading to an os.Exit(1) in the end.
+	err := runAbapLandscapePortalUpdateAddOnProduct(&config, client)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runAbapLandscapePortalUpdateAddOnProduct(config *abapLandscapePortalUpdateAddOnProductOptions, client http.Client) error {
+	var systemId, reqId, reqStatus string
+	var clientAT http.Client
+	var servKey serviceKey
+	var getStatusReq http.Request
+	var err error
+
+	if prepareErr := parseServiceKeyAndPrepareAccessTokenHttpClient(config.LandscapePortalAPIServiceKey, &clientAT, &servKey); prepareErr != nil {
+		err = fmt.Errorf("Failed to prepare credentials to get access token of LP API. Error: %v\n", prepareErr)
+		return err
+	}
+
+	// get system
+	if getSystemErr := getSystemBySystemNumber(config, client, clientAT, servKey, &systemId); getSystemErr != nil {
+		err = fmt.Errorf("Failed to get system with systemNumber %v. Error: %v\n", config.AbapSystemNumber, getSystemErr)
+		return err
+	}
+
+	// update addon in the system
+	if updateAddOnErr := updateAddOn(config, client, clientAT, servKey, systemId, &reqId); updateAddOnErr != nil {
+		err = fmt.Errorf("Failed to update addon in the system with systemId %v. Error: %v\n", systemId, updateAddOnErr)
+		return err
+	}
+
+	// prepare http request to poll status of addon update
+	if prepareGetStatusHttpRequestErr := prepareGetStatusHttpRequest(clientAT, servKey, reqId, &getStatusReq); prepareGetStatusHttpRequestErr != nil {
+		err = fmt.Errorf("Failed to prepare http request to poll status of addon update request %v. Error: %v\n", reqId, prepareGetStatusHttpRequestErr)
+		return err
+	}
+
+	// query status of request
+	if pollStatusOfUpdateAddOnErr := pollStatusOfUpdateAddOn(client, &getStatusReq, reqId, &reqStatus); pollStatusOfUpdateAddOnErr != nil {
+		err = fmt.Errorf("Failed to poll status of addon update request %v. Error: %v\n", reqId, pollStatusOfUpdateAddOnErr)
+		return err
+	}
+
+	// keep polling status of addon update request until it reaches a final status (C/E/X)
+	for reqStatus == StatusInProgress || reqStatus == StatusScheduled {
```

Comment:
You do not have any timeout. If the backend call dumps the for loop will run endless. (ok, only unit the pipeline run over all times out but this could be probabbly too long and you might want to execute some logic upon failure. golang has a special for flavor for timeouts, here an example
func (br *buildWithRepository) waitToBeFinished(maxRuntimeInMinutes time.Duration, pollInterval time.Duration) error {
	timeout := time.After(maxRuntimeInMinutes)
	ticker := time.Tick(pollInterval)
	for {
		select {
		case <-timeout:
			return errors.Errorf("Timed out: (max Runtime %v reached)", maxRuntimeInMinutes)
		case <-ticker:
			if err := br.build.Get(); err != nil {
				return err
			}
			if !br.build.IsFinished() {
				log.Entry().Infof("Assembly of %s is not yet finished, check again in %s", br.repo.PackageName, pollInterval)
			} else {
				return nil
			}
		}
	}
}